modules/platform/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/idempotency/NuxeoIdempotentFilter.java

```diff
@@ -0,0 +1,168 @@
+ * <li>return the stored response if a subsequent request with the same key is processed again
+ * <li>return a conflict response if a request with the same key is processed while the first request is still in
+ * progress.
+ * </ul>
+ *
+ * @since 11.5
+ */
+public class NuxeoIdempotentFilter implements Filter {
+
+    private static final Logger log = LogManager.getLogger(NuxeoIdempotentFilter.class);
+
+    public static final String HEADER_KEY = "Idempotency-Key";
+
+    public static final String STORE_PROPERTY = "org.nuxeo.request.idempotency.keyvaluestore.name";
+
+    public static final String DEFAULT_STORE = "idempotentrequest";
+
+    protected static final long DEFAULT_TTL_SECONDS = Duration.ofDays(1).toSeconds();
+
+    public static final String TTL_SECONDS_PROPERTY = "org.nuxeo.request.idempotency.ttl.seconds";
+
+    public static final String INPROGRESS_MARKER = "IDEMPOTENCY_INPROGRESS_MARKER";
+
+    public static final String INFO_SUFFIX = "_info";
+
+    protected static final int DEFERRED_OUTPUT_STREAM_THRESHOLD = 1024 * 1024; // 1 MB
+
+    // safe methods according to RFC 7231 4.2.1
+    protected static final Set<String> SAFE_METHODS = Set.of(HttpGet.METHOD_NAME, HttpHead.METHOD_NAME,
+            HttpOptions.METHOD_NAME, HttpTrace.METHOD_NAME);
+
+    // idempotent methods according to RFC 7231 4.2.2
+    protected static final Set<String> IDEMPOTENT_METHODS = Set.of(HttpPut.METHOD_NAME, HttpDelete.METHOD_NAME);
+
+    protected long getTTLSeconds() {
+        ConfigurationService cs = Framework.getService(ConfigurationService.class);
+        if (cs != null) {
+            return cs.getLong(TTL_SECONDS_PROPERTY, DEFAULT_TTL_SECONDS);
+        }
+        return DEFAULT_TTL_SECONDS;
+    }
+
+    protected String getStoreName() {
+        ConfigurationService cs = Framework.getService(ConfigurationService.class);
+        if (cs != null) {
+            return cs.getString(STORE_PROPERTY, DEFAULT_STORE);
+        }
+        return DEFAULT_STORE;
+    }
+
+    protected boolean isIdempotentMethod(String method) {
+        return SAFE_METHODS.contains(method) || IDEMPOTENT_METHODS.contains(method);
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+        String method = request.getMethod();
+        if (isIdempotentMethod(method)) {
+            log.debug("No idempotent processing done: method is already idempotent: " + method);
+            chain.doFilter(request, response);
+            return;
+        }
+        String key = request.getHeader(NuxeoIdempotentFilter.HEADER_KEY);
+        if (key == null) {
+            log.debug("No idempotent processing done: no Idempotent-Key present");
+            chain.doFilter(request, response);
```

Comment:
I'd prefer all the repeated `chain.doFilter(request, response)` in a caller method. So rename the current method to `doFilterIdempotent` and then have:

    @Override 
    public void doFilter(...) throws ... {
        if (!doFilterIdempotent()) {
            chain.doFilter(request, response);
        }
    }

And return `false` from `doFilterIdempotent` whenever you haven't done any processing.